pyspi/octaveless_config.yaml

```diff
@@ -0,0 +1,963 @@
+
+#     - estimator: kozachenko
+#       history_length: 1
+
+#     - estimator: kernel
+#       history_length: 1
+
+#     - estimator: gaussian
+#       history_length: 10
+
+#     - estimator: kozachenko
+#       history_length: 10
+
+#     - estimator: kernel
+#       history_length: 10
+
+#   DirectedInfo: # No theiler window yet
+#     - estimator: gaussian
+
+#     - estimator: kozachenko
+
+#     - estimator: kernel
+
+#   StochasticInteraction: # No theiler window
+#     - estimator: gaussian
+
+#     - estimator: kozachenko
+
+#     - estimator: kernel
+
+#   # Mutual information
+#   MutualInfo:
+#     - estimator: gaussian
+
+#     - estimator: kraskov
+#       prop_k: 4
+
+#     - estimator: kraskov
+#       prop_k: 4
+#       dyn_corr_excl: AUTO
+
+#     - estimator: kernel
+#       kernel_width: 0.25
+
+#   # Mutual information
+#   TimeLaggedMutualInfo:
+#     - estimator: gaussian
+
+#     - estimator: kraskov
+#       prop_k: 4
+
+#     - estimator: kraskov
+#       prop_k: 4
+#       dyn_corr_excl: AUTO
+
+#     - estimator: kernel
+#       kernel_width: 0.25
+
+#   # Transfer entropy
+#   TransferEntropy:
+#     # Kraskov estimator with auto-embedding on source/target and DCE
+#     - estimator: kraskov
+#       prop_k: 4
+#       auto_embed_method: MAX_CORR_AIS
+#       k_search_max: 10
+#       tau_search_max: 4
+
+#     - estimator: kraskov
+#       prop_k: 4
+#       auto_embed_method: MAX_CORR_AIS
+#       k_search_max: 10
+#       tau_search_max: 4
+#       dyn_corr_excl: AUTO
+
+#     # Kraskov estimator with auto-embedding on target-only, src history of 1, and DCE
+#     - estimator: kraskov
+#       prop_k: 4
+#       k_history: 2
+#       l_history: 1
+#       dyn_corr_excl: AUTO
+
+#     # Kraskov estimator with fixed embedding of 1 for source/target and DCE
+#     - estimator: kraskov
+#       prop_k: 4
+#       k_history: 1
+#       l_history: 1
+#       dyn_corr_excl: AUTO
+
+#     # Same as above with no DCE
+#     - estimator: kraskov
+#       prop_k: 4
+#       k_history: 1
+#       l_history: 1
+
+#     # Currently the Kernel method has an overload issue with the AIS calculator..
+#     # Kernel estimator with auto-embedding on source/target and DCE
+#     # - estimator: kernel
+#     #   auto_embed_method: MAX_CORR_AIS
+#     #   k_search_max: 4
+#     #   kernel_width: 0.25
+
+#     # Kernel estimator with no auto-embedding on source/target and DCE
+#     - estimator: kernel
+#       kernel_width: 0.25
+#       k_history: 1
+#       l_history: 1
+
+#     # Gaussian estimator doesn't have DCE (aka Bartlett corrections) yet
+#     - estimator: gaussian
+#       auto_embed_method: MAX_CORR_AIS
+#       k_search_max: 10
+#       tau_search_max: 2
+
+#     - estimator: gaussian
+#       k_history: 1
+#       l_history: 1
+
+#     - estimator: symbolic
+#       k_history: 1
+#       l_history: 1
+
+#     - estimator: symbolic
+#       k_history: 10
+#       l_history: 1
+
+#   IntegratedInformation:

```

Comment:
@jmoo2880 from memory only the `IntegratedInformation` SPIs would need to be commented out to get it to work without octave. See here: https://github.com/DynamicsAndNeuralSystems/pyspi/blob/a0f7167ca19f6033bb1ece836d09a80101cbf0e8/pyspi/statistics/infotheory.py#L568C26-L568C26

Reply:
@olivercliff Ah yes, I had initially arrived at this .yaml file by sequentially removing each Information Theoretic SPI until the "Octave missing" error vanished. The motivation for this "octaveless" version was to allow users to skip the computation of SPIs with octave dependence, and continue with a reduced set of the remaining SPIs. 
Upon closer inspection, it does appear that only the IIT SPIs call on oct2py/octave, but the initialisation of the SPIs from the infotheory module will terminate when attempting to import octave (in the event it doesn't exist), rendering the remainder of the (octave independent) SPIs inaccessible. This explains why commenting out the entire info theory module (and not just IIT SPIs) works without throwing any errors relating to octave.